data/src/main/java/com/microsoft/azure/kusto/data/KustoColumn.java

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.azure.kusto.data;
+
+/**
+ * Represents a single Kusto result column with a type, name and nullability information
+ * 
+ * @param <C>
+ *            the Java class returned by the ADX column type
+ * @param <T>
+ *            type of ADX column
+ */
+public class KustoColumn<C, T extends KustoType<C>> {
```

Comment:
oalso if you would have had the 'ordinal' and could invoke getObject(ordinal) instead and save all the map lookups

Reply:
1. Where's the "KustoColumn" type? I didn't see anything on a GitHub search except this PR (https://github.com/Azure/azure-kusto-java/search?q=KustoColumn)

2. How do you propose we get the ordinal information? If we detect it automatically and cache it, the mapper instance will only work correctly with queries that return the columns in the same exact order. That's a foot gun. Or do you mean we add a constructor that takes an ordinal instead of a column name so it's provided by the user?

Reply:
1. https://github.com/Azure/azure-kusto-java/blob/a1417109a264ca56ac7d56adaae0095ea460d7b0/data/src/main/java/com/microsoft/azure/kusto/data/KustoResultColumn.java
2. You cache it per KustoResultSetTable which is anyway different for each query you run ...

Reply:
1. Done, I merged KustoColumn and QueryResultColumnInfo into a new "ObjectPopulator" class
2. Done. `extractList` now caches the column's ordinals per invocation. And users can manually specify ordinals in the mappers instead (or on top of) column names.

Reply:
Note about point 2 - currently `KustoResultSetTable` doesn't implement the `ResultSet` correctly in ways that affect this change (see issue #161). For example, `ResultSet` ordinals are 1-based, while `KustoResultSetTable` uses 0-based numbering.